test/src/navigation.spec.ts

```diff
@@ -719,7 +719,7 @@ describe('navigation', function () {
           waitEvent(page, 'frameattached').then(_frame => {
             return (frame = _frame);
           }),
-          waitEvent(page, 'framenavigated', f => {
+          waitEvent(page, 'framenavigated').then(f => {
```

Comment:
Why was this changed?
This now does not wait for the correct frame to be navigated but any frame to be navigated.

Reply:
Oh, this extra argument acts like a filter. I wasn't aware of that. So I'll revert this part and check why the test is failing.